lib/api/documentExtractor.js

```diff
@@ -26,6 +26,27 @@ const kerror = require('../kerror');
 const assertionError = kerror.wrap('api', 'assert');
 
 const extractors = ([
+  {
+    methods: {
+      extractFromRequest: request => [
+        {
+          _id: request.input.args._id,
+          fields: request.input.body.fields,
```

Comment:
Deleted fields should be in `_source`, for example you could recreate an object from lodash path with properties set to `undefined`

Reply:
I don't really understand what you want here? I did this, is this okay ? :
```js
      insertInRequest: ([documents], request) => {
        request.input.args._id = documents._id;
        request.input.body.fields = documents._source;
        return request;
      },
```